src/App.tsx

```diff
@@ -1,14 +1,108 @@
 /* eslint-disable max-len */
-import React from 'react';
+// #region import
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { getTodos, getUser } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+// #endregion import
+
 export const App: React.FC = () => {
+  // #region State
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [openTodoId, setOpenTodoId] = useState<number | null>(null);
+
+  const [todosLoading, setTodosLoading] = useState(false);
+  const [modalLoading, setModalLoading] = useState(false);
+
+  const [filterValue, setFilterValue] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
+
+  // eslint-disable-next-line no-console
+  console.log('Render');
+
+  useEffect(() => {
+    setTodosLoading(true);
+
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .finally(() => setTodosLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (!currentTodo || currentTodo.user) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(user => {
+        setCurrentTodo(prevTodo => {
+          if (prevTodo?.id === currentTodo.id) {
+            return { ...prevTodo, user };
+          }
+
+          return prevTodo;
+        });
+      })
+      .finally(() => setModalLoading(false));
+
+    // // eslint-disable-next-line no-console
+    // console.log('Modal state:', isModalOpen, 'ModalLoader:', modalLoading);
+  }, [currentTodo]);
+
+  // #endregion State
+
+  // #region Handler
+
+  const handleOpenModal = (todo: Todo): void => {
+    setOpenTodoId(todo.id);
+    setModalLoading(true);
+    setIsModalOpen(true);
+
+    setCurrentTodo(todo);
+  };
+
+  const filteredAndSearchedTodos = useMemo(() => {
+    return todos
+      .filter(todo => {
+        if (filterValue === 'completed') {
+          return todo.completed === true;
+        } else if (filterValue === 'active') {
```

Comment:
Create a enum for 'all', 'active', 'completed' and use it everywhere. Also, will be better if you use switch case here instead of if conditions